src/delivery/git/mod.rs

```diff
@@ -279,27 +280,38 @@ pub fn create_repo(path: &PathBuf) -> Result<(), DeliveryError> {
     }
 }
 
-pub fn create_or_update_delivery_remote(url: &str, path: &PathBuf) -> Result<bool, DeliveryError> {
-    let result = git_command(&["remote", "add", "delivery", &url], path);
-    match result {
-        Ok(_) => return Ok(true),
-        Err(e) => {
-            match e.detail.clone() {
-                Some(msg) => {
-                    if msg.contains("remote delivery already exists") {
-                        try!(git_command(&["remote", "rm", "delivery"], path));
-                        try!(git_command(&["remote", "add", "delivery", &url], path));
-                        return Ok(false)
-                    } else {
-                        return Err(e)
-                    }
-                },
-                None => {
-                    return Err(e)
-                }
+// Returns the (Git) delivery remote URL form the specified repository path
+//
+// ex.=> ssh://user@ent@delivery.example.com:8989/ent/organization/foo
+pub fn delivery_remote_from_repo<P>(path: P) -> DeliveryResult<String>
+        where P: AsRef<Path> {
+    git_command(&["config", "--get", "remote.delivery.url"], path.as_ref())
+        .map(|g| g.stdout.trim().to_string())
+        // If there is no 'delivery' remote, return an empty String
+        .or(Ok(String::from("")))
+}
+
+// Update the (Git) delivery remote
+//
+// Try to add the delivery remote and if it fails adding it, try to remove
+// it first and then add it. This way we ensure we are updating it with the
+// provided URL no mather if it already exists or not.
+pub fn update_delivery_remote<P, S>(url: S, path: P) -> DeliveryResult<()>
+        where P: AsRef<Path>,
+              S: AsRef<str> {
+    let path = path.as_ref();
+    let url  = url.as_ref();
+    git_command(&["remote", "add", "delivery", url], path)
+        .map(|_| ())
```

Comment:
An observation: The previous code relied on using `match` while this code uses some of the more sophisticated methods available for working with Result types.  The result is less deep-nesting at the expense of requiring a bit more rust knowledge on the part of the reader.

Not really sure which I prefer so I wouldn't advocate changing it, just thought I would point it out.